src/main/java/it/gov/pagopa/pu/classification/service/BalanceAmountService.java

```diff
@@ -0,0 +1,80 @@
+
+@Service
+@Slf4j
+public class BalanceAmountService {
+
+  private final BalanceService balanceService;
+  private final BalanceDefaultMarshallingService balanceDefaultMarshallingService;
+  private final ScriptEngineManager factory = new ScriptEngineManager();
+  private final ScriptEngine engine = factory.getEngineByName("rhino");
+
+  public BalanceAmountService(BalanceService balanceService, BalanceDefaultMarshallingService balanceDefaultMarshallingService) {
+    this.balanceService = balanceService;
+    this.balanceDefaultMarshallingService = balanceDefaultMarshallingService;
+  }
+
+  public String calculateAmountBalance(CalculateAmountBalanceRequest calculateAmountBalanceRequest) {
+    String balance = calculateAmountBalanceRequest.getBalance();
+    BigDecimal amountInstallment = Utilities.longCentsToBigDecimalEuro(calculateAmountBalanceRequest.getAmountCents());
+
+    Object balanceXML = balanceService.unmarshalBalance(balance);
+
+    if (balanceXML instanceof CtBilancio) {
+      log.info("The balance amount is already calculated");
+      return balance;
+    }
+
+    try {
+      CtBilancioDefault ctBilancioDefault = (CtBilancioDefault) balanceXML;
+      log.info("Calculating balance amount resolving default type");
+      for (CtCapitoloDefault capitolo : ctBilancioDefault.getCapitolo()) {
+        for (CtAccertamentoDefault ctAccertamentoDefault : capitolo.getAccertamento()) {
+          BigDecimal calculatedAmount;
+          if (ctAccertamentoDefault.getImporto().equals(BalanceDefaultAmountType.TOTAL.getType())) {
+            calculatedAmount = amountInstallment;
+          } else if (ctAccertamentoDefault.getImporto().contains(BalanceDefaultAmountType.EXTRACT_AMOUNT.getType())) {
+            engine.eval(ctAccertamentoDefault.getImporto());
+            Invocable invocable = (Invocable) engine;
+            String result = String.valueOf(invocable.invokeFunction(BalanceDefaultAmountType.EXTRACT_AMOUNT.getType(),
+              calculateAmountBalanceRequest.getRemittanceInformation()));
+            calculatedAmount = new BigDecimal(result);
+          } else if (ctAccertamentoDefault.getImporto().contains(BalanceDefaultAmountType.CALCULATE_AMOUNT.getType())) {
+            engine.eval(ctAccertamentoDefault.getImporto());
+            Invocable invocable = (Invocable) engine;
+            String result = String.valueOf(
+              invocable.invokeFunction(BalanceDefaultAmountType.CALCULATE_AMOUNT.getType(), amountInstallment));
+            calculatedAmount = new BigDecimal(result);
+          } else {
+            throw new InvalidValueException(ctAccertamentoDefault.getImporto() + " as function type to calculate amount balance not supported");
+          }
+          String amountString = Utilities.amountToString(calculatedAmount);
+          String amountWithoutSeparator = amountString.replace("\\.", "");
+          String amountWithSeparatorAndDecimal = amountWithoutSeparator.replace(",", ".");
```

Comment:
if we need this format, let's modify the amountToString format with #.00 instead of made these replacements next